lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,802 @@
+  my $t = Test::Mojo->new('Frogs');
+
+This object initializes a L<Mojo::UserAgent> object, loads the Mojolicious
+application C<Frogs>, binds and listens on a free TCP port (e.g., 32114), and
+starts the application event loop. When the L<Test::Mojo> object (C<$t>) goes
+out of scope, the application is stopped.
+
+Relative URLs in the test object method assertions (C<get_ok>, C<post_ok>,
+etc.) will be sent to the Mojolicious application started by L<Test::Mojo>:
+
+  # Rewritten to http://localhost:32114/frogs
+  $t->get_ok('/frogs');
+
+L<Test::Mojo> has a lot of handy shortcuts built into it to make testing
+L<Mojolicious> or L<Mojolicious::Lite> applications enjoyable.
+
+=head3 An example
+
+Let's spin up a Mojolicious application using C<mojo generate app MyApp>. The
+C<mojo> utility will create a working application and a C<t> directory with a
+working test file:
+
+  $ mojo generate app MyApp
+  [mkdir] /my_app/script
+  [write] /my_app/script/my_app
+  [chmod] /my_app/script/my_app 744
+  ...
+  [mkdir] /my_app/t
+  [write] /my_app/t/basic.t
+  ...
+
+Let's run the tests (we'll create the C<log> directory to quiet the
+application output):
+
+  $ cd my_app
+  $ mkdir log
+  $ prove -lv t
+  t/basic.t ..
+  ok 1 - GET /
+  ok 2 - 200 OK
+  ok 3 - content is similar
+  1..3
+  ok
+  All tests successful.
+  Files=1, Tests=3,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.33 cusr  0.07 csys =  0.44 CPU)
+  Result: PASS
+
+The boilerplate test file looks like this:
+
+  use Mojo::Base -strict;
+
+  use Test::More;
+  use Test::Mojo;
+
+  my $t = Test::Mojo->new('MyApp');
+  $t->get_ok('/')->status_is(200)->content_like(qr/Mojolicious/i);
+
+  done_testing();
+
+Here we can see our application class name C<MyApp> is passed to the
+L<Test::Mojo> constructor. Under the hood, L<Test::Mojo> creates a new
+L<Mojo::Server> instance, loads C<MyApp> (which we just created), and runs the
+application. We write our tests with relative URLs because L<Test::Mojo> takes
+care of getting the request to the running test application (since its port
+may change between runs).
+
+=head3 Testing with configuration data
+
+We can alter the behavior of our application using environment variables (such
+as C<MOJO_MODE>) and through configuration values. One nice feature of
+L<Test::Mojo> is its ability to pass configuration values directly from its
+constructor.
+
+Let's modify our application and add a "feature flag" to enable a new feature
+when the C<enable_weather> configuration value is set:
+
+  # Load configuration from hash returned by "my_app.conf"
+  my $config = $self->plugin('Config');
+
+  # Normal route to controller
+  $r->get('/')->to('example#welcome');
+
+  # NEW: this route only exists when 'enable_weather' is set in the configuration
+  if ($config->{enable_weather}) {
+      $r->get('/weather')->to(cb => sub { shift->render(text => "It's hot! 🔥") }
+  }
+
+To test this new feature, we don't even need to create a configuration file—we
```

Comment:
I'm using an em-dash here… I couldn't find an example in the existing docs of this. Happy to change the wording or use a colon.